kep/20200903-node-resources-allocatable-scoring.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Node resources allocatable scoring
+authors:
+  - "@patricklundquist"
+owning-sig: sig-scheduling
+reviewers:
+  - "@Huang-Wei"
+  - "@ahg-g"
+approvers:
+  - "@Huang-Wei"
+creation-date: 2020-09-03
+last-updated: 2020-09-03
+status: provisional
+---
+
+# Node resources allocatable scoring
+
+## Table of Contents
+
+<!-- toc -->
+- [Node resources allocatable scoring](#node-resources-allocatable-scoring)
+  - [Table of Contents](#table-of-contents)
+  - [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+  - [Terms](#terms)
+  - [Proposal](#proposal)
+  - [Design Details](#design-details)
+    - [Extension points](#extension-points)
+      - [Score](#score)
+  - [Known Limitations](#known-limitations)
+  - [Alternatives considered](#alternatives-considered)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Testing Plan](#testing-plan)
+  - [Implementation History](#implementation-history)
+  - [References](#references)
+<!-- /toc -->
+
+## Motivation
+A Kubernetes cluster can consist of nodes with widely varying resource capacities. Likewise, workloads may consist of pods with widely varying resource requests. A pod with high resource requests might only fit on one of the largest capacity nodes. If one or more small pods are scheduled on the largest capacity nodes, the large pod may be unscheduleable. That is a sub-optimal scheduling when there are smaller nodes available that could have taken the small pods. The existing scheduling priorities don't offer a solution to this scenario. To improve cluster utilization, this proposal tries to solve the problem by implementing a scoring plugin ```NodeResourcesAllocatable``` based on the scheduler framework.
+ 
```

Comment:
We may add a few sentences to explain the shortage of existing `NodeResourcesLeastAllocated` and `NodeResourcesMostAllocated` - which uses **ratio** of allocated resources. While in this plugin, if I understanding it correctly, it's using the **absolute number** of allocatable resources to score/normalize, which quite varies with the in-tree Score plugins.

Reply:
Done